cumulusci/tasks/create_package_version.py

```diff
@@ -609,7 +609,10 @@ def _convert_project_dependencies(self, dependencies):
         new_dependencies = []
         for dependency in dependencies:
             new_dependency = {}
-            if isinstance(dependency, PackageVersionIdDependency):
+            if isinstance(dependency, PackageVersionIdDependency) or (
+                isinstance(dependency, PackageNamespaceVersionDependency)
+                and dependency.version_id
+            ):
```

Comment:
Would it make more sense to simply do a structural check for the presence of version_id and not also the isinstance checks?

btw you can pass a tuple as the 2nd parameter of isinstance